src/actuator/ActuatorBulkFAST.C

```diff
@@ -0,0 +1,308 @@
+        const int offset = turbIdOffset_.h_view(iTurb);
+        auto epsilonChord = Kokkos::subview(
+          actMeta.epsilonChord_.view_host(), iTurb, Kokkos::ALL);
+        auto epsilonRef =
+          Kokkos::subview(actMeta.epsilon_.view_host(), iTurb, Kokkos::ALL);
+        auto epsilonTower = Kokkos::subview(
+          actMeta.epsilonTower_.view_host(), iTurb, Kokkos::ALL);
+
+        for (int np = 0; np < numForcePts; np++) {
+
+          auto epsilonLocal =
+            Kokkos::subview(epsilon_.view_host(), np + offset, Kokkos::ALL);
+          auto epsilonOpt =
+            Kokkos::subview(epsilonOpt_.view_host(), np + offset, Kokkos::ALL);
+
+          switch (openFast_.getForceNodeType(iTurb, np)) {
+          case fast::HUB: {
+            float nac_cd = openFast_.get_nacelleCd(iTurb);
+            // Compute epsilon only if drag coefficient is greater than zero
+            if (nac_cd > 0) {
+              float nac_area = openFast_.get_nacelleArea(iTurb);
+
+              // This model is used to set the momentum thickness
+              // of the wake (Martinez-Tossas PhD Thesis 2017)
+              float tmpEps = std::sqrt(2.0 / M_PI * nac_cd * nac_area);
+              for (int i = 0; i < 3; i++) {
+                epsilonLocal(i) = tmpEps;
+              }
+            }
+            // If no nacelle force just specify the standard value
+            // (it will not be used)
+            else {
+              for (int i = 0; i < 3; i++) {
+                epsilonLocal(i) = epsilonRef(i);
+              }
+            }
+            for (int i = 0; i < 3; i++) {
+              epsilonOpt(i) = epsilonLocal(i);
+            }
+            break;
+          }
+          case fast::BLADE: {
+            double chord = openFast_.getChord(np, iTurb);
+            for (int i = 0; i < 3; i++) {
+              // Define the optimal epsilon
+              epsilonOpt(i) = epsilonChord(i) * chord;
+              epsilonLocal(i) = std::max(epsilonOpt(i), epsilonRef(i));
+            }
+            break;
+          }
+          case fast::TOWER: {
+            for (int i = 0; i < 3; i++) {
+              epsilonLocal(i) = epsilonTower(i);
+              epsilonOpt(i) = epsilonLocal(i);
+            }
+            break;
+          }
+          default:
+            throw std::runtime_error("Actuator line model node type not valid");
+            break;
+          }
+
+          for (int i = 0; i < 3; ++i) {
+            ThrowAssertMsg(
+              epsilonLocal(i) > 0.0,
+              "Epsilon zero for point: " + std::to_string(np) + " index " +
+                std::to_string(i));
+          }
+
+          // The radius of the searching. This is given in terms of
+          //   the maximum of epsilon.x/y/z/.
+          searchRadius_.h_view(np + offset) =
+            std::max(
+              epsilonLocal(0), std::max(epsilonLocal(1), epsilonLocal(2))) *
+            sqrt(log(1.0 / 0.001));
```

Comment:
I'm not sure if this is left over from the original code, but what's up with the factor of ~2.63 on epsilon 1 and 2 in the search radius.

Reply:
Yeah not sure.  I was thinking it was a spherical Gaussian factor but never did the math.  Maybe @tonyinme knows?  I'm also going to get rid of the extra division and just put in 1000.0.

Reply:
Yes, this is meant to apply the Gaussian up to the point where it's value is .1% of what it is in the center. 
The formula being solved is: 1 / exp(-r^2/epsilon^2) = .001
You can get rid of the division. It would be good to add a comment of where it comes from, something like:
"This is the length where the value of the Gaussian becomes 0.1 % (1.0 / .001 = 1000) of the value at the center of the Gaussian"

Reply:
Okay I'll add the comment. Thanks @tonyinme!